avatars/processors/inter_record_range_difference_test.py

```diff
@@ -0,0 +1,182 @@
+        new_range_variable_name="range_value",
+        new_difference_variable_name="value_difference",
+    )
+
+    processed_df = processor.preprocess(df_with_cumulated)
+    print(processed_df)
+
+    assert_frame_equal(processed_df, preprocessed_df_with_cumulated, check_dtype=False)
+
+
+def test_postprocess(
+    df_with_cumulated: pd.DataFrame, preprocessed_df_with_cumulated: pd.DataFrame
+) -> None:
+    "Verify that the postprocess is correct."
+    processor = InterRecordRangeDifferenceProcessor(
+        id_variable="id",
+        target_start_variable="start",
+        target_end_variable="end",
+        sort_by_variable_name="start",
+        new_first_variable_name="first_value",
+        new_range_variable_name="range_value",
+        new_difference_variable_name="value_difference",
+    )
+
+    postprocessed_df = processor.postprocess(
+        df_with_cumulated, preprocessed_df_with_cumulated
+    )
+    print(postprocessed_df)
+    assert_frame_equal(postprocessed_df, df_with_cumulated, check_dtype=False)
+
+
+def test_preprocess_raises_error_when_missing_values(
+    df_with_cumulated: pd.DataFrame,
+) -> None:
+    df_with_cumulated.loc[0, "end"] = np.nan
+    processor = InterRecordRangeDifferenceProcessor(
+        id_variable="id",
+        target_start_variable="start",
+        target_end_variable="end",
+        sort_by_variable_name="start",
+        new_first_variable_name="first_value",
+        new_range_variable_name="range_value",
+        new_difference_variable_name="value_difference",
+    )
+
+    with pytest.raises(ValueError, match="Expected no missing values for"):
+        processor.preprocess(df_with_cumulated)
+
+
+def test_preprocess_raises_error_when_wrong_var(
+    df_with_cumulated: pd.DataFrame,
+) -> None:
+    processor = InterRecordRangeDifferenceProcessor(
+        id_variable="id",
+        target_start_variable="start",
+        target_end_variable="end",
+        sort_by_variable_name="wrong_start",
+        new_first_variable_name="first_value",
+        new_range_variable_name="range_value",
+        new_difference_variable_name="value_difference",
+    )
+
+    with pytest.raises(ValueError, match="Expected valid variable names for"):
+        processor.preprocess(df_with_cumulated)
+
+
+def test_postprocess_raises_error_when_wrong_var(
+    df_with_cumulated: pd.DataFrame, preprocessed_df_with_cumulated: pd.DataFrame
+) -> None:
+    processor = InterRecordRangeDifferenceProcessor(
+        id_variable="id",
+        target_start_variable="start",
+        target_end_variable="end",
+        sort_by_variable_name="start",
+        new_first_variable_name="first_value",
+        new_range_variable_name="wrong_range_value",
+        new_difference_variable_name="value_difference",
+    )
+
+    with pytest.raises(ValueError, match="Expected valid variable names for"):
+        processor.postprocess(df_with_cumulated, preprocessed_df_with_cumulated)
```

Comment:
Nit: All of this error checking could have been done with a `pytest.parametrize` like so. Same with the missing_values one:

```python
@pytest.parametrize("argument_name", "sort_by_variable_name", ["new_first_variable_name", "new_range_variable_name"])
def test_postprocess_raises_error_when_wrong_var(argument_name: str):
    # all the correct arguments, that should pass without errors
    arguments = dict(
        id_variable="id",
        target_start_variable="start",
        target_end_variable="end",
        sort_by_variable_name="start",
        new_first_variable_name="first_value",
        new_range_variable_name="range_value",
        new_difference_variable_name="value_difference")
	
    # assign a wrong value to one of the argument
    arguments[argument_name] = "wrong_value"
 
	InterRecordRangeDifferenceProcessor(**arguments)

    with pytest.raises(ValueError, match="Expected valid variable names for"):
        processor.postprocess(df_with_cumulated, preprocessed_df_with_cumulated)
```